docs/architecture/transaction.md

```diff
@@ -1,56 +1,143 @@
 
-## What is the purpose of a transaction?
+![Transaction diagram](../img/architecture/transaction/transaction-diagram.png)
 
-In Miden, a `Transaction` represents the state transition of a single account. A `Transaction` takes a single [account](accounts.md) and zero or more [notes](notes.md), and none or one script (piece of code executed after all notes have been executed) as input, and outputs the same account with a potentially updated state, together with some potential newly created notes.
+Compared to most blockchains, where a `Transaction` typically involves more than one account (e.g., sender and receiver), a `Transaction` in Miden involves a single account. To illustrate, Alice sends 5 ETH to Bob. In Miden, sending 5 ETH from Alice to Bob takes two `Transaction`s, one in which Alice creates a note containing 5 ETH and one in which Bob consumes that note and receives the 5 ETH. This model removes the need for a global lock on the blockchain's state, enabling Miden to process `Transaction`s in parallel.
 
-![Transaction diagram](../img/architecture/transaction/transaction-diagram.png)
+## Transaction lifecycle
+
+Every `Transaction` describes the process of an account changing its state. This process is described as a Miden VM program, resulting in the generation of a zero-knowledge proof.
+
+![Transaction execution process](../img/architecture/transaction/transaction-execution-process.png)
+
+### Transaction execution flow
+
+`Transaction`s are being executed in a well-defined sequence, in which several note and transaction scripts can call the account interface to interact with it.
+
+![Transaction execution flow](../img/architecture/transaction/transaction-program.png)
+
+1. **Prologue**: On-chain commitments are validated against provided data.
+2. **Note processing**: Input notes are executed sequentially against the account, following a  sequence defined by the executor.
+    - Notes must be consumed fully (all assets must be transferred to the account or to another note).
+    - The note script must be executed in full with the provided note inputs and transaction arguments. Transaction arguments can be injected by the executor for each note at runtime.
+3. **Transaction script execution (optional)**: A `Transaction` script, if present, is executed.
+    - This script can sign the `Transaction` or directly interact with the account without using notes.
+4. **Epilogue**: Execution completes, resulting in an updated account state and a generated zero-knowledge proof. The validity of the state change is being checked
+    - The account state is updated.
+    - New notes are created (optional), transferring assets from the account to the newly created notes.
+    - The `Nonce` must be incremented, to allow transaction authentication. Also the net sum of all involved assets mus be 0 - if the account is no faucet.
```

Comment:
We should re-write this as a sequence of paragraphs rather than a bullet-point list.

Also, I think in some ways this is too detailed (e.g., I'm not sure we need to bring up the destination between transaction inputs and args at this point) but also is missing a lot of relevant data. For example, I think it would be interesting to note that in the prologue we verify that all input notes have been previously created (except for the ephemeral notes).

Overall, each of these bullet points should be 1 - 3 paragraphs explaining the basic things that happens and why they happen.

Reply:
I changed it into a sequence of paragraphs that describes in which order things are happening and why.